data/netcdf_data.py

```diff
@@ -757,32 +757,42 @@ def get_nc_file_list(self, datasetconfig: DatasetConfig, **kwargs: dict) -> Unio
             except KeyError:
                 raise RuntimeError(
                     "Opening a dataset via sqlite requires the 'variable' keyword argument.")
-            if not isinstance(variable, list):
-                variable = [variable]
+            if not isinstance(variable, set):
+                variable = set(variable)
 
             calculated_variables = datasetconfig.calculated_variables
-            if variable[0] in calculated_variables:
-                equation = calculated_variables[variable[0]]['equation']
-
-                variable = data.utils.get_data_vars_from_equation(
-                    equation, [v.key for v in db.get_data_variables()])
+            variables_to_load = self.__get_variables_to_load(db, variable, calculated_variables)
 
             try:
                 timestamp = self.__get_requested_timestamps(
-                    db, variable[0], kwargs['timestamp'], kwargs.get('endtime'), kwargs.get('nearest_timestamp', False))
+                    db, variables_to_load[0], kwargs['timestamp'], kwargs.get('endtime'), kwargs.get('nearest_timestamp', False))
```

Comment:
the timestamp and endtime kwargs are the same for all variables when loading, so just grab the first variable.